src/libs/blueprint/blueprint_mesh.cpp

```diff
@@ -74,34 +125,1720 @@ namespace conduit
+                       Node &info)
+{
+    bool res = true;
+
+    std::string proto_name = "mesh::coordset";
+    
+    if(!coordset.has_child("type"))
+    {
+        log_error(info,proto_name, "missing child \"type\"");
+        res = false;
+    }
+    else if(!coordset::type::verify(coordset["type"],info["type"]))
+    {
+        res = false;
+    }
+    else
+    {
+        std::string type_name = coordset["type"].as_string();
+
+        if(type_name == "uniform")
+        {
+            res = mesh::coordset::uniform::verify(coordset,info);
+        }
+        else if(type_name == "rectilinear")
+        {
+            res = mesh::coordset::rectilinear::verify(coordset,info);
+        }
+        else if(type_name == "explicit")
+        {
+            res = mesh::coordset::_explicit::verify(coordset,info);
+        }
+    }
+
+    log_verify_result(info,res);
+
+    return res;
+}
+
 //-----------------------------------------------------------------------------
-// -- end conduit::blueprint::mesh --
+// blueprint::mesh::coordset::type protocol interface
 //-----------------------------------------------------------------------------
 
+//-----------------------------------------------------------------------------
+bool
+mesh::coordset::type::verify(const Node &coordset_type,
+                             Node &info)
+{    
+    bool res = true;
+    std::string proto_name = "mesh::coordset::type";
+
+
+    if(!coordset_type.dtype().is_string())
+    {
+        log_error(info,proto_name,"is not a string");
+        res = false;
+    }
+    else
+    {
+        std::string coordset_type_str = coordset_type.as_string();
+
+        if(coordset_type_str == "uniform"     ||
+           coordset_type_str == "rectilinear" ||
+           coordset_type_str == "explicit")
+        {
+            log_info(info,proto_name,"valid type: " + coordset_type_str);
+        }
+        else
+        {
+            log_error(info,
+                      proto_name,
+                      "unsupported value:" + coordset_type_str);
+            res = false;
+        }
+    }
+
+    log_verify_result(info,res);
+
+    return res;
+}
+
+//-----------------------------------------------------------------------------
+// blueprint::mesh::coordset::coord_system protocol interface
+//-----------------------------------------------------------------------------
+
+//-----------------------------------------------------------------------------
+bool
+check_cart_coord_sys_axis_name(std::string &name)
+{
+    return ( name == "x" || name == "y" || name == "z");
+}
+
+//-----------------------------------------------------------------------------
+bool
+check_sph_coord_sys_axis_name(std::string &name)
+{
+    return ( name == "r" || name == "theta" || name == "phi");
+}
+
+//-----------------------------------------------------------------------------
+bool
+check_cyln_coord_sys_axis_name(std::string &name)
+{
+    return ( name == "r" || name == "z");
```

Comment:
rz is for 2D cylindrical, the "r" component collapses 2 comps from 3d. 
 
all the sim codes I know about only support 2D-cylindrical -- for the 3D case the r would be broken out into two components.